### PR TITLE
[Keyboard] Add soda/mango

### PR DESCRIPTION
**Pull request by gezhaoyou**
## Description

 Add new keyboard soda/mango
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by zvecr** [comment]
You should push changes to the same branch, and not close/open new PRs.

File: keyboards/soda/mango/config.h
Comment (zvecr) [inline keyboards/soda/mango/config.h:1]: This file should be removed.

File: keyboards/soda/mango/mango.h
Comment (zvecr) [inline keyboards/soda/mango/mango.h:1]: This file should be removed.

**Review by zvecr — CHANGES_REQUESTED** [review]
From `qmk lint`:
```
linting soda/mango
⚠ soda/mango: Layout "LAYOUT" is offset on X axis by 3.25
⚠ soda/mango: Layout "LAYOUT" is offset on Y axis by 1.5
⚠ soda/mango: Layout "LAYOUT" is offset on X axis by 3.25
⚠ soda/mango: Layout "LAYOUT" is offset on Y axis by 1.5
Ψ Lint check passed!
⚠ soda/mango: Layout "LAYOUT" is offset on X axis by 3.25
Keyboard Name: Mango
⚠ soda/mango: Layout "LAYOUT" is offset on Y axis by 1.5
Manufacturer: Soda
Website: https://github.com/gezhaoyou
Maintainer: JeayoKeh
Layouts: LAYOUT
Processor: STM32F072
Bootloader: stm32-dfu
Layout Layout:
             ┌──┐┌──┐┌──┐┌──┐
             │  ││  ││  ││  │
             └──┘└──┘└──┘└──┘
             ┌──┐┌──┐┌──┐┌──┐
             │  ││  ││  ││  │
             └──┘└──┘└──┘│  │
             ┌──┐┌──┐┌──┐│  │
             │  ││  ││  ││  │
             └──┘└──┘└──┘└──┘
             ┌──┐┌──┐┌──┐
             │  ││  ││  │
             └──┘└──┘└──┘
             ┌──────┐┌──┐┌──┐
             │      ││  ││  │
             └──────┘└──┘│  │
                         │  │
                         │  │
                         └──┘
```

File: keyboards/soda/mango/mango.h
Comment (gezhaoyou) [reply keyboards/soda/mango/mango.h:1]: ok, i have remove it.

File: keyboards/soda/mango/config.h
Comment (gezhaoyou) [reply keyboards/soda/mango/config.h:1]: ok, i have remove it.

**Comment by gezhaoyou** [comment]
> From `qmk lint`:
> 
> ```
> linting soda/mango
> ⚠ soda/mango: Layout "LAYOUT" is offset on X axis by 3.25
> ⚠ soda/mango: Layout "LAYOUT" is offset on Y axis by 1.5
> ⚠ soda/mango: Layout "LAYOUT" is offset on X axis by 3.25
> ⚠ soda/mango: Layout "LAYOUT" is offset on Y axis by 1.5
> Ψ Lint check passed!
> ⚠ soda/mango: Layout "LAYOUT" is offset on X axis by 3.25
> Keyboard Name: Mango
> ⚠ soda/mango: Layout "LAYOUT" is offset on Y axis by 1.5
> Manufacturer: Soda
> Website: https://github.com/gezhaoyou
> Maintainer: JeayoKeh
> Layouts: LAYOUT
> Processor: STM32F072
> Bootloader: stm32-dfu
> Layout Layout:
>              ┌──┐┌──┐┌──┐┌──┐
>              │  ││  ││  ││  │
>              └──┘└──┘└──┘└──┘
>              ┌──┐┌──┐┌──┐┌──┐
>              │  ││  ││  ││  │
>              └──┘└──┘└──┘│  │
>              ┌──┐┌──┐┌──┐│  │
>              │  ││  ││  ││  │
>              └──┘└──┘└──┘└──┘
>              ┌──┐┌──┐┌──┐
>              │  ││  ││  │
>              └──┘└──┘└──┘
>              ┌──────┐┌──┐┌──┐
>              │      ││  ││  │
>              └──────┘└──┘│  │
>                          │  │
>                          │  │
>                          └──┘
> ```

I have modify my info.json  and now no warning

File: keyboards/soda/mango/mango.c
Comment (zvecr) [inline keyboards/soda/mango/mango.c:1]: This file should be removed.

File: keyboards/soda/mango/mcuconf.h
Comment (zvecr) [inline keyboards/soda/mango/mcuconf.h:1]: This file should be removed.

File: keyboards/soda/mango/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+ [0] = LAYOUT(
```
Comment (zvecr) [inline keyboards/soda/mango/keymaps/via/keymap.c:20]:
```suggestion
    [0] = LAYOUT_numpad_5x4(
```

File: keyboards/soda/mango/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        KC_NUM_LOCK,            KC_PSLS,    KC_PAST,        LT(1, KC_PMNS),
+        KC_P7,                  KC_P8,      KC_P9,
+        KC_P4,                  KC_P5,      KC_P6,          KC_PPLS,
+        KC_P1,                  KC_P2,      KC_P3,
+                                KC_P0,      KC_PDOT,        KC_PENT),
+   [1] = LAYOUT(
```
Comment (zvecr) [inline keyboards/soda/mango/keymaps/via/keymap.c:26]:
```suggestion
    [1] = LAYOUT_numpad_5x4(
```

File: keyboards/soda/mango/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT(
```
Comment (zvecr) [inline keyboards/soda/mango/keymaps/default/keymap.c:20]:
```suggestion
    [0] = LAYOUT_numpad_5x4(
```

File: keyboards/soda/mango/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        KC_NUM_LOCK,            KC_PSLS,    KC_PAST,        LT(1, KC_PMNS),
+        KC_P7,                  KC_P8,      KC_P9,
+        KC_P4,                  KC_P5,      KC_P6,          KC_PPLS,
+        KC_P1,                  KC_P2,      KC_P3,
+                                KC_P0,      KC_PDOT,        KC_PENT),
+   [1] = LAYOUT(
```
Comment (zvecr) [inline keyboards/soda/mango/keymaps/default/keymap.c:26]:
```suggestion
    [1] = LAYOUT_numpad_5x4(
```

File: keyboards/soda/mango/info.json [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        "rows": ["B10", "B2", "B1", "B0", "A7"]
+    },
+    "usb": {
+        "device_version": "1.0.0",
+        "force_nkro": true,
+        "vid": "0x3A54",
+        "pid": "0x4F5D"
+    },
+    "layouts": {
+        "LAYOUT": {
```
Comment (zvecr) [inline keyboards/soda/mango/info.json:20]:
```suggestion
        "LAYOUT_numpad_5x4": {
```

File: keyboards/soda/mango/info.json [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        "cols": ["A6", "A5", "A4", "A3"],
+        "rows": ["B10", "B2", "B1", "B0", "A7"]
+    },
+    "usb": {
+        "device_version": "1.0.0",
+        "force_nkro": true,
+        "vid": "0x3A54",
+        "pid": "0x4F5D"
+    },
+    "layouts": {
```
Comment (zvecr) [inline keyboards/soda/mango/info.json:19]:
```suggestion
    "community_layouts": ["numpad_5x4"],
    "layouts": {
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/soda/mango/readme.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+```shell
+qmk compile -kb soda/mango -km default
+```
+
+**via:**
+
+```shell
+qmk compile -kb soda/mango -km via
+```
```
Comment (waffle87) [inline keyboards/soda/mango/readme.md:20]:
```suggestion
Make example for this keyboard (after setting up your build environment):

    make soda/mango:default

Flashing example for this keyboard:

    make soda/mango:default:flash
```

**Review by waffle87 — APPROVED** [review]

File: keyboards/soda/mango/info.json [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    "matrix_pins": {
+        "cols": ["A6", "A5", "A4", "A3"],
+        "rows": ["B10", "B2", "B1", "B0", "A7"]
+    },
+    "usb": {
+        "device_version": "1.0.0",
+        "force_nkro": true,
+        "vid": "0x3A54",
+        "pid": "0x4F5D"
+    },
```
Comment (zvecr) [inline keyboards/soda/mango/info.json:18]:
```suggestion
    },
    "features": {
        "bootmagic": true,
        "extrakey": true,
        "mousekey": true,
        "nkro": true
    },
```

File: keyboards/soda/mango/readme.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
+# mango-17
+
+A customizable 17  keyboard, support both HOTSWAP and SOLDER.
+
+* Keyboard Maintainer: [gezhaoyou](https://github.com/gezhaoyou)
+* Hardware Supported: [gezhaoyou](https://github.com/gezhaoyou)
+
+Make example for this keyboard (after setting up your build environment):
+
```
Comment (zvecr) [inline keyboards/soda/mango/readme.md:9]:
```suggestion
```

File: keyboards/soda/mango/readme.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+    make soda/mango:default:flash
+
+## Bootloader
+
+Enter the bootloader in 2 ways:
+
+* **Physical reset button**: Briefly press the button: [boot] first, then press button: [reset]  on the back of the PCB
+* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```
Comment (zvecr) [inline keyboards/soda/mango/readme.md:25]:
```suggestion
See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).

## Bootloader

Enter the bootloader in 2 ways:

* **Physical reset button**: Briefly press the button: [boot] first, then press button: [reset]  on the back of the PCB
* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by zvecr — APPROVED** [review]

**Pull request merged**